File: data/EXP01/golang_no_system_var/2671/2671.output_3.go
type FrequencyTracker struct {
    data map[int]int
}

func Constructor() FrequencyTracker {
    return FrequencyTracker{
        data: make(map[int]int),
    }
}

func (this *FrequencyTracker) Add(number int) {
    this.data[number]++
}

func (this *FrequencyTracker) DeleteOne(number int) {
    if this.data[number] > 0 {
        this.data[number]--
    }
}

func (this *FrequencyTracker) HasFrequency(frequency int) bool {
    for _, count := range this.data {
        if count == frequency {
            return true
        }
    }
    return false
}

File: data/EXP01/python/2671/2671.output_3.py
# @lc app=leetcode id=2671 lang=python3
class FrequencyTracker:

    def __init__(self):
        self.data = {}

    def add(self, number: int) -> None:
        if number in self.data:
            self.data[number] += 1
        else:
            self.data[number] = 1

    def deleteOne(self, number: int) -> None:
        if number in self.data:
            self.data[number] -= 1
            if self.data[number] == 0:
                del self.data[number]

    def hasFrequency(self, frequency: int) -> bool:
        return frequency in self.data.values()